build/utils/yarn.go

```diff
@@ -183,6 +184,15 @@ func buildYarnV1DependencyMap(packageInfo *PackageInfo, responseStr string, allo
 		if err != nil {
 			return
 		}
+		if packageCleanName == "" || packageVersion == "" {
+			// If we enter this case it means we got an unexpected name for a dependency that might cause issues later on when constructing the tree
+			log.Debug(fmt.Sprintf("got an unexpected dependency name when building Yarn V1 dependency map.\nfull name: '%s' | clean name: '%s' | clean version: '%s'", curDependency.Name, packageCleanName, packageVersion))
+			if allowPartialResults {
+				log.Warn(fmt.Sprintf("got an unexpected package name during Yarn V1 dependencies map calculation: %s\nFinal rasults may be partial", curDependency.Name))
+				continue
+			}
+			err = fmt.Errorf("couldn't parse correctly the following dependency during Yarn V1 dependencies map calculation: %s", curDependency.Name)
```

Comment:
Shouldn't there be a return here after creating this error?
If not, then what's the poing of creating this error?

Reply:
Correct, I missed adding the return